assembly/src/library/module.rs

```diff
@@ -0,0 +1,75 @@
+use alloc::vec::Vec;
+
+use super::LibraryPath;
+use crate::{
+    ast::{ProcedureIndex, ProcedureName},
+    RpoDigest,
+};
+
+// MODULE INFO
+// ================================================================================================
+
+#[derive(Debug, Clone)]
+pub struct ModuleInfo {
+    path: LibraryPath,
+    procedures: Vec<ProcedureInfo>,
+}
+
+impl ModuleInfo {
+    /// Returns a new [`ModuleInfo`] instantiated library path.
+    pub(super) fn new(path: LibraryPath) -> Self {
```

Comment:
I think this should be public, so that other library wrappers can construct this information.

Reply:
Done!